exercises/poker/.meta/src/reference/java/Hand.java

```diff
@@ -2,108 +2,129 @@
+				.collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (e1, e2) -> e1, LinkedHashMap::new));
+		List<Integer> rankCounts = frequencyMap.entrySet().stream().map(Map.Entry::getValue)
+				.collect(Collectors.toList());
+		List<Integer> suits = cards.stream().map(Card::getSuit).collect(Collectors.toList());
 
-        return calculatedScore(frequencyMap, cardsByRank, ranks, rankCounts, suits);
-    }
+		return calculatedScore(frequencyMap, cardsByRank, ranks, rankCounts, suits);
+	}
 
-    private int calculatedScore(Map<Integer, Integer> frequencyMap, List<Card> cardsByRank, List<Integer> ranks,
-                                List<Integer> rankCounts, List<Integer> suits) {
-        if (ranks.equals(Arrays.asList(14, 5, 4, 3, 2))) {
-            ranks = Arrays.asList(5, 4, 3, 2, 1);
-        }
+	private int calculatedScore(Map<Integer, Integer> frequencyMap, List<Card> cardsByRank, List<Integer> ranks,
+			List<Integer> rankCounts, List<Integer> suits) {
+		if (ranks.equals(Arrays.asList(14, 5, 4, 3, 2))) {
+			ranks = Arrays.asList(5, 4, 3, 2, 1);
+		}
 
-        boolean flush = suits.stream().distinct().count() == 1;
-        boolean straight = ranks.stream().distinct().count() == 5 && ranks.get(0) - ranks.get(4) == 4;
-        Iterator<Integer> iteratorOverFrequencies = frequencyMap.keySet().iterator();
-        int highestFrequency = iteratorOverFrequencies.next();
+		boolean flush = suits.stream().distinct().count() == 1;
+		boolean straight = ranks.stream().distinct().count() == 5 && ranks.get(0) - ranks.get(4) == 4;
+		Iterator<Integer> iteratorOverFrequencies = frequencyMap.keySet().iterator();
+		int highestFrequency = iteratorOverFrequencies.next();
+
+		int maxValue = Collections.max(ranks);
+
+		if (straight && flush) {
+			return 800 + highestFrequency;
+		}
+		if (rankCounts.equals(Arrays.asList(4, 1))) {
+			return 700 + cardsByRank.get(0).getRank();
+		}
+		if (rankCounts.equals(Arrays.asList(3, 2))) {
+			int triplet = 0;
+			int pair = 0;
+			for (Object key : frequencyMap.keySet()) {
+				if (frequencyMap.get(key) == 2) {
+					pair = (int) key;
+				}
+				if (frequencyMap.get(key) == 3) {
+					triplet = 3 * (int) key;
+				}
+			}
+			return 600 + 3 * triplet + pair;
+		}
+		if (flush) {
+			return 500 + highestFrequency;
+		}
+		if (straight) {
+			return 400 + maxValue;
+		}
+		if (rankCounts.equals(Arrays.asList(3, 1, 1))) {
+			List<Integer> cardsf1 = new ArrayList<Integer>();
+			int triplet = 0;
+			for (Object key : frequencyMap.keySet()) {
+				if (frequencyMap.get(key) == 1) {
+					cardsf1.add((int) key);
+				}
+				if (frequencyMap.get(key) == 3) {
+					triplet = 3 * (int) key;
+				}
+			}
+
+			return 300 + triplet + Collections.max(cardsf1);
+		}
+		if (rankCounts.equals(Arrays.asList(2, 2, 1))) {
+			int sum = 0;
+			for (Object key : frequencyMap.keySet()) {
+				int fKey = (int) key;
+				int fValue = frequencyMap.get(key);
+				sum += fKey * fValue;
```

Comment:
Again, `key` should probably have type `Integer` not `Object`. And try to avoid abbreviations in names, as they often make it harder to understand what the variable is. Instead of `fKey` and `fValue`, `frequencyKey` and `frequencyValue` (or just `frequency`) would be more descriptive 🙂 

And what is `sum` a sum over? Try to include that in the name if possible 🙂